app/code/core/Mage/Core/Model/Resource/Config.php

```diff
@@ -133,7 +133,7 @@ public function loadToXml(Mage_Core_Model_Config $xmlConfig, $condition = null)
         $deleteStores = [];
         // set stores config values from database
         foreach ($rowset as $r) {
-            if ($r['scope'] !== 'stores') {
+            if (is_null($r['value']) || $r['scope'] !== 'stores') {
```

Comment:
Didn't test, but if (is_null($r['value']) then $value = empty string in PHP7, and it would continue to line 40. Shouldn't PHP8 do the same?

Reply:
From my tests the array could look like

```
$r = [
    ...
    'value' => null
];
```

We can leave this open until the errors that popup before are fixed. Could be easier to test.

Reply:
Can you please test it?

Your right, this changes some behavoir ...

### Test
- set some config to null in DB (i've used Title Prefix form Admin -> Design config)
- load/save config
- set debug breakpoint at `Varien_Simplexml_Config::setNode()`

Reply:
I'll look into it after my holiday.

Reply:
> and it would continue to line

Right, but there we just create an empty XML-node for a `null` DB value. 

With this change the node is not created. What could happen? Access a not-existing node (`getConfigNode()` or `getNode()`)? Did not found any errors yet.

I could be an improvement to not create emtpy xml-nodes, but needs some testing ...

I'll open a new PR with casting to string to move forward. This can be investigated later. :)

Reply:
@elidrissidev tested this and everything looks fine. (Method description says _Load configuration values into xml config object_ and we just avoid loading `null` to config.)

Reply:
> Default valuew from config.xml are only used when no config path exists in `core_config_data`. It is not used when a `null` entry exists in DB.

@sreichel where did you that it's not used if value in db is null?

Reply:
Tested it.

Reply:
I have tested this by setting `contacts/email/recipient_email` to `NULL` in db (it has default value of "hello@ example.com" in xml), and then running:
```php
Mage::app()->getConfig()->getNode('default/contacts/email/recipient_email')
```
I could see that before adding the `is_null($r['value'])` check, it was returning an empty string value, whereas after that it's returning the value from xml. So might not be totally safe.

Reply:
Thanks. Tested `getConfigNode()` at the beginning - w/o default config value.